### ISSUE TITLE
vfs-http-adapter's /directory/ vs /file

### ISSUE THREAD

**Issue by orlin** [c9/vfs#26]
Do trailing slashes have to be required to get a directory through vfs-http-adapter?  It shouldn't matter what the thing is - vfs knows it's there and I'd like to have it (with a `200` ok) anyway.  Currently a `500` "EISDIR: Requested resource is a directory".  Or check for preference with `mountOptions`?

**Comment by creationix** [comment]
Recently I ported this code to my moonslice project and what I did there was to redirect instead of spitting out a 500 error.  I really want only one cannonical representation for a directory, but I understand that people don't always include the trailing slash properly.  For example, go to http://luvit.io/modules/web/middleware and you'll be redirected with the trailing slash added.

I'll take a pull request to do the redirect like I do in my other code. https://github.com/luvit/web-static/blob/master/static.lua#L137-143 and https://github.com/luvit/web-static/blob/master/static.lua#L122-127 and https://github.com/luvit/web-static/blob/master/static.lua#L84-88

**Comment by orlin** [comment]
Redirecting is ok, if it's in response to (handling) the error.  Would it be possible to configure vfs in such a way (the custom errors issue)?  

The reason I filed this one is that `flatiron/api-easy` just strips the ending slash - it can't test.  If I was typing it, I'd do the same.  In my implementation case, 

Otherwise I'd just append the slash (through middleware) if the resource is without extension (check with regex) or explicit accept header for content-type.  Better than further hitting the fs, preemptively...

**Comment by creationix** [comment]
The thing is that in vfs-http-adapter I treat urls ending in slashes as directories and others as files.  If you want some other semantic then make another module.  If I see a request for "/foo" and know that "/foo" is a directory, it would be violation of my semantics to serve the directory resource at that url.  But responding with a redirect to the correct url is good because then the second response is something I can respond to.

In the past (like in Connect) I responded without a redirect and it caused issues where relative links would change meaning depending on if there was a trailing slash and other issues.

You're free to modify this module to have the semantics you want and use yours instead.  It is liberally licensed after all.  I did make is separate from vfs itself on purpose.
